packages/resources/src/RemixSite.ts

```diff
@@ -0,0 +1,1506 @@
+   *
+   * @default false
+   */
+   edge?: boolean;
+
+  /**
+   * Path to the directory where the website source is located.
+   */
+  path: string;
+
+  /**
+   * The customDomain for this website. SST supports domains that are hosted
+   * either on [Route 53](https://aws.amazon.com/route53/) or externally.
+   *
+   * Note that you can also migrate externally hosted domains to Route 53 by
+   * [following this guide](https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/MigratingDNS.html).
+   *
+   * @example
+   * ```js {3}
+   * new RemixSite(stack, "RemixSite", {
+   *   path: "path/to/site",
+   *   customDomain: "domain.com",
+   * });
+   * ```
+   *
+   * ```js {3-6}
+   * new RemixSite(stack, "RemixSite", {
+   *   path: "path/to/site",
+   *   customDomain: {
+   *     domainName: "domain.com",
+   *     domainAlias: "www.domain.com",
+   *     hostedZone: "domain.com"
+   *   },
+   * });
+   * ```
+   */
+  customDomain?: string | RemixDomainProps;
+
+  /**
+   * An object with the key being the environment variable name.
+   *
+   * @example
+   * ```js {3-6}
+   * new RemixSite(stack, "RemixSite", {
+   *   path: "path/to/site",
+   *   environment: {
+   *     API_URL: api.url,
+   *     USER_POOL_CLIENT: auth.cognitoUserPoolClient.userPoolClientId,
+   *   },
+   * });
+   * ```
+   */
+  environment?: Record<string, string>;
+
+  /**
+   * When running `sst start`, a placeholder site is deployed. This is to ensure
+   * that the site content remains unchanged, and subsequent `sst start` can
+   * start up quickly.
+   *
+   * @example
+   * ```js {3}
+   * new RemixSite(stack, "RemixSite", {
+   *   path: "path/to/site",
+   *   disablePlaceholder: true,
+   * });
+   * ```
+   */
+  disablePlaceholder?: boolean;
+
+  defaults?: {
+    function?: {
+      timeout?: number;
+      memorySize?: number;
+      permissions?: Permissions;
+    };
+  };
+
+  /**
+   * While deploying, SST waits for the CloudFront cache invalidation process to finish. This ensures that the new content will be served once the deploy command finishes. However, this process can sometimes take more than 5 mins. For non-prod environments it might make sense to pass in `false`. That'll skip waiting for the cache to invalidate and speed up the deploy process.
+   */
+  waitForInvalidation?: boolean;
+}
+
+/**
+ * The `RemixSite` construct is a higher level CDK construct that makes it easy to create a Remix app. It provides a simple way to build and deploy the site to an S3 bucket; setup a CloudFront CDN for fast content delivery; and configure a custom domain for the website URL.
+ *
+ * By default, the Remix app server is deployed to a Lambda function behind API Gateway HTTP API Lambda. Alternatively, you can choose to deploy to Lambda@Edge.
+ * 
+ * In the case you have a centralized database, Edge locations are often far away from your database. If you are making multiple queries in your data loader, you might experience long latency.
```

Comment:
**suggestion:** Perhaps this warrants one of those `:::caution` / warning wrappers? Perhaps even a section of its own?

Reply:
I believe @thdxr is doing a stream on this topic next week? We could possibly even link to that from here.